### DISCUSSION TITLE
Compiling supabase-wrappers failing

### DISCUSSION THREAD

**Discussion by breynolds-nvenioanalytics** [supabase/supabase#27222] [Questions]
Full disclosure, I'm new to Supabase and Rust.  I'm trying to compile and deploy s3_fdw to PostgreSQL 13 and am not sure what to do with these errors.  They appear to be in the supabase wrapper?

Compiling supabase-wrappers v0.1.18 (/usr/local/src/wrappers/supabase-wrappers)
error[E0425]: cannot find function `add_row_identity_var` in crate `pg_sys`
   --> supabase-wrappers/src/modify.rs:102:25
    |
102 |                 pg_sys::add_row_identity_var(root, var, rtindex, &attr.attname.data as _);
    |                         ^^^^^^^^^^^^^^^^^^^^ not found in `pg_sys`

error[E0308]: mismatched types
   --> supabase-wrappers/src/interface.rs:649:56
    |
649 |             fdw_routine.AddForeignUpdateTargets = Some(modify::add_foreign_update_targets);
    |                                                   ---- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ incorrect number of function parameters
    |                                                   |
    |                                                   arguments to this enum variant are incorrect
    |
    = note: expected fn pointer `unsafe extern "C" fn(*mut pgrx::pg_sys::Query, *mut RangeTblEntry, *mut RelationData)`
                  found fn item `extern "C" fn(*mut PlannerInfo, u32, *mut RangeTblEntry, *mut RelationData) {add_foreign_update_targets}`
help: the type constructed contains `extern "C" fn(*mut PlannerInfo, u32, *mut RangeTblEntry, *mut RelationData) {add_foreign_update_targets}` due to the type of the argument passed
   --> supabase-wrappers/src/interface.rs:649:51
    |
649 |             fdw_routine.AddForeignUpdateTargets = Some(modify::add_foreign_update_targets);
    |                                                   ^^^^^----------------------------------^
    |                                                        |
    |                                                        this argument influences the type of `Some`
note: tuple variant defined here
   --> /builddir/build/BUILD/rustc-1.75.0-src/library/core/src/option.rs:577:5

**Comment by GaryAustin1** [comment]
Don't know much about it, but....
![image](https://github.com/supabase/supabase/assets/54564956/fb65154c-08ca-48dd-b9bd-80b689748640)
https://github.com/supabase/wrappers
